services/src/main/java/org/keycloak/crypto/RsaCekManagementProvider.java

```diff
@@ -39,6 +40,8 @@ public JWEAlgorithmProvider jweAlgorithmProvider() {
             jcaAlgorithmName = "RSA/ECB/PKCS1Padding";
         } else if (JWEConstants.RSA_OAEP.equals(jweAlgorithmName)) {
             jcaAlgorithmName = "RSA/ECB/OAEPWithSHA-1AndMGF1Padding";
+        } else if (JWEConstants.RSA_OAEP_256.equals(jweAlgorithmName)) {
+            return new RsaKeyEncryption256JWEAlgorithmProvider("RSA/ECB/OAEPWithSHA-256AndMGF1Padding");
```

Comment:
[4.3.  Key Encryption with RSAES OAEP of JWA](https://tools.ietf.org/html/rfc7518#section-4.3) states that
> "alg" Param Value : RSA-OAEP-256
> Key Management Algorithm : RSAES OAEP using SHA-256 and MGF1 with SHA-256 

[A.2.  Key Management Algorithm Identifier Cross-Reference of JWA](A.2.  Key Management Algorithm Identifier Cross-Reference) stats that
> JWE : RSA-OAEP-256
> JCA : RSA/ECB/OAEPWithSHA-256AndMGF1Padding & MGF1ParameterSpec.SHA256

I think RSA-OAEP-256 use SHA-256 for both the main digest and the MGF1 digest.
I'm not sure 
`return new RsaKeyEncryption256JWEAlgorithmProvider("RSA/ECB/OAEPWithSHA-256AndMGF1Padding")`
can instanciate the provider for satisfying RSA-OAEP-256. (might be use SHA-256 for the main digest and SHA-1 for the MGF1 digest?).

Reply:
This is handled in `org.keycloak.jose.jwe.alg.RsaKeyEncryption256JWEAlgorithmProvider` where the `AlgorithmParameterSpec` is passed in

```
AlgorithmParameterSpec paramSpec = new OAEPParameterSpec("SHA-256", "MGF1", MGF1ParameterSpec.SHA256, PSource.PSpecified.DEFAULT);
algp.init(paramSpec);
```

There isn't a more specific cipher than "RSA/ECB/OAEPWithSHA-256AndMGF1Padding" available - passing in those `AlgorithmParameterSpec` seems to be the only way to get SHA-256 on the main and MFG1 digest.

Reply:
@dillon-sellars Thank you. I've confirmed the point you've mentioned.